jenkins/aws/buildOpenapi.sh

```diff
@@ -0,0 +1,158 @@
+                    "${AUTOMATION_BUILD_DIR}/../../**/*spec/${BUILD_DIR}/openapi.yml" \
+                    "${AUTOMATION_BUILD_DIR}/../../**/*spec/${BUILD_DIR}/openapi.yaml" \
+                    "${AUTOMATION_BUILD_DIR}/../../**/*spec/${BUILD_DIR}/swagger.json" \
+                    "${AUTOMATION_BUILD_DIR}/../../**/*spec/${BUILD_DIR}/swagger.yml" \
+                    "${AUTOMATION_BUILD_DIR}/../../**/*spec/${BUILD_DIR}/swagger.yaml" \
+                    "${AUTOMATION_BUILD_DIR}/../../../**/*spec/${BUILD_DIR}/openapi.json" \
+                    "${AUTOMATION_BUILD_DIR}/../../../**/*spec/${BUILD_DIR}/openapi.yml" \
+                    "${AUTOMATION_BUILD_DIR}/../../../**/*spec/${BUILD_DIR}/openapi.yaml" \
+                    "${AUTOMATION_BUILD_DIR}/../../../**/*spec/${BUILD_DIR}/swagger.json" \
+                    "${AUTOMATION_BUILD_DIR}/../../../**/*spec/${BUILD_DIR}/swagger.yml" \
+                    "${AUTOMATION_BUILD_DIR}/../../../**/*spec/${BUILD_DIR}/swagger.yaml" \
+                    "${AUTOMATION_BUILD_DIR}/openapi.json" \
+                    "${AUTOMATION_BUILD_DIR}/openapi.yml" \
+                    "${AUTOMATION_BUILD_DIR}/openapi.yaml" \
+                    "${AUTOMATION_BUILD_DIR}/swagger.json" \
+                    "${AUTOMATION_BUILD_DIR}/swagger.yml" \
+                    "${AUTOMATION_BUILD_DIR}/swagger.yaml" \
+                    "${AUTOMATION_BUILD_DIR}/../**/*spec/openapi.json" \
+                    "${AUTOMATION_BUILD_DIR}/../**/*spec/openapi.yml" \
+                    "${AUTOMATION_BUILD_DIR}/../**/*spec/openapi.yaml" \
+                    "${AUTOMATION_BUILD_DIR}/../**/*spec/swagger.json" \
+                    "${AUTOMATION_BUILD_DIR}/../**/*spec/swagger.yml" \
+                    "${AUTOMATION_BUILD_DIR}/../**/*spec/swagger.yaml" \
+                    "${AUTOMATION_BUILD_DIR}/../../**/*spec/openapi.json" \
+                    "${AUTOMATION_BUILD_DIR}/../../**/*spec/openapi.yml" \
+                    "${AUTOMATION_BUILD_DIR}/../../**/*spec/openapi.yaml" \
+                    "${AUTOMATION_BUILD_DIR}/../../**/*spec/swagger.json" \
+                    "${AUTOMATION_BUILD_DIR}/../../**/*spec/swagger.yml" \
+                    "${AUTOMATION_BUILD_DIR}/../../**/*spec/swagger.yaml" \
+                    "${AUTOMATION_BUILD_DIR}/../../../**/*spec/openapi.json" \
+                    "${AUTOMATION_BUILD_DIR}/../../../**/*spec/openapi.yml"\
+                    "${AUTOMATION_BUILD_DIR}/../../../**/*spec/openapi.yaml"\
+                    "${AUTOMATION_BUILD_DIR}/../../../**/*spec/swagger.json" \
+                    "${AUTOMATION_BUILD_DIR}/../../../**/*spec/swagger.yml" \
+                    "${AUTOMATION_BUILD_DIR}/../../../**/*spec/swagger.yaml")
+
+# Was a spec file found?
+[[ ! -f "${OPENAPI_SPEC_FILE}" ]] && fatal "Can't find source openAPI file" && exit 1
+
+# Allow for the bundle context to be broadened
+if [[ -n "${BUNDLE_CONTEXT_DIR}" ]]; then
```

Comment:
Couldn't we put this above the search then limit the findFiles to within the context dir?

Reply:
Yep we could - I thought of that but it was a more major change given the variety of paths and the sequencing between them. Because all the searching is done in one go, the desired spec should be found within the bundle context so it felt ok to check after.

However as you asked, I'll take a look at dynamically building the find list.